boa/src/builtins/map/mod.rs

```diff
@@ -0,0 +1,328 @@
+
+        this.set_internal_slot(INSTANCE_PROTOTYPE, prototype);
+        // This value is used by console.log and other routines to match Object type
+        // to its Javascript Identifier (global constructor method name)
+
+        // add our arguments in
+        let data = match args.len() {
+            0 => OrderedMap::new(),
+            _ => match &args[0] {
+                Value::Object(object) => {
+                    let object = object.borrow();
+                    if let Some(map) = object.as_map_clone() {
+                        map
+                    } else if object.is_array() {
+                        let mut map = OrderedMap::new();
+                        let len = i32::from(&args[0].get_field("length"));
+                        for i in 0..len {
+                            let val = &args[0].get_field(i.to_string());
+                            let (key, value) = Self::get_key_value(val).ok_or_else(|| {
+                                ctx.construct_type_error(
+                                    "iterable for Map should have array-like objects",
+                                )
+                            })?;
+                            map.insert(key, value);
+                        }
+                        map
+                    } else {
+                        return Err(ctx.construct_type_error(
+                            "iterable for Map should have array-like objects",
+                        ));
+                    }
+                }
+                _ => {
+                    return Err(
+                        ctx.construct_type_error("iterable for Map should have array-like objects")
+                    )
+                }
+            },
+        };
+
+        // finally create length property
+        Self::set_size(this, data.len());
+
+        this.set_data(ObjectData::Map(GcCell::new(data)));
+
+        Ok(this.clone())
+    }
+
+    /// Create a new `Map` object.
+    pub(crate) fn create(global: &Value) -> Value {
+        // Create prototype
+        let prototype = Value::new_object(Some(global));
+
+        make_builtin_fn(Self::set, "set", &prototype, 2);
+        make_builtin_fn(Self::delete, "delete", &prototype, 1);
+        make_builtin_fn(Self::get, "get", &prototype, 1);
+        make_builtin_fn(Self::clear, "clear", &prototype, 0);
+        make_builtin_fn(Self::has, "has", &prototype, 1);
+        make_builtin_fn(Self::for_each, "forEach", &prototype, 1);
+
+        make_constructor_fn(
+            Self::NAME,
+            Self::LENGTH,
+            Self::make_map,
+            global,
+            prototype,
+            true,
+        )
+    }
+
+    /// Initialise the `Map` object on the global object.
+    #[inline]
+    pub(crate) fn init(global: &Value) -> (&str, Value) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        (Self::NAME, Self::create(global))
+    }
```

Comment:
```suggestion
    /// Initialise the `Map` object on the global object.
    pub(crate) fn init(global: &Value) -> Value {
    	let _timer = BoaProfiler::global().start_event(Self::NAME, "init");

        // Create prototype
        let prototype = Value::new_object(Some(global));

        make_builtin_fn(Self::set, "set", &prototype, 2);
        make_builtin_fn(Self::delete, "delete", &prototype, 1);
        make_builtin_fn(Self::get, "get", &prototype, 1);
        make_builtin_fn(Self::clear, "clear", &prototype, 0);
        make_builtin_fn(Self::has, "has", &prototype, 1);
        make_builtin_fn(Self::for_each, "forEach", &prototype, 1);

        let map_object = make_constructor_fn(
            Self::NAME,
            Self::LENGTH,
            Self::make_map,
            global,
            prototype,
            true,
        );
        
        (Self::NAME, map_object)
    }
```
#547 Marged `create` into `init` because init was always empty and we did not need an extra function.

Reply:
Sorry, `init` should return a `(&str, Value)` :sweat_smile: